src/Components/RecsListTable/index.js

```diff
@@ -0,0 +1,503 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { useIntl } from 'react-intl';
+import { Link } from 'react-router-dom';
+
+import {
+  Table,
+  TableBody,
+  TableHeader,
+  TableVariant,
+} from '@patternfly/react-table';
+import {
+  Card,
+  CardBody,
+  Pagination,
+  PaginationVariant,
+  Stack,
+  Tooltip,
+  TooltipPosition,
+} from '@patternfly/react-core';
+
+import { DateFormat } from '@redhat-cloud-services/frontend-components/DateFormat';
+import { InsightsLabel } from '@redhat-cloud-services/frontend-components/InsightsLabel';
+import { Main } from '@redhat-cloud-services/frontend-components/Main';
+import PrimaryToolbar from '@redhat-cloud-services/frontend-components/PrimaryToolbar/PrimaryToolbar';
+
+import {
+  FILTER_CATEGORIES as FC,
+  RECS_LIST_COLUMNS,
+  RULE_CATEGORIES,
+  TOTAL_RISK_LABEL_LOWER,
+} from '../../AppConstants';
+import { useGetRecsQuery } from '../../Services/SmartProxy';
+import messages from '../../Messages';
+import {
+  RECS_LIST_INITIAL_STATE,
+  updateRecsListFilters as updateFilters,
+} from '../../Services/Filters';
+import RuleLabels from '../RuleLabels/RuleLabels';
+import { strong } from '../../Utilities/intlHelper';
+import Loading from '../Loading/Loading';
+import { ErrorState, NoMatchingRecs } from '../MessageState/EmptyStates';
+import RuleDetails from '../Recommendation/RuleDetails';
+
+const RecsListTable = () => {
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const filters = useSelector(({ filters }) => filters.recsListState);
+  const { isError, isUninitialized, isFetching, isSuccess, data } =
+    useGetRecsQuery({ impacting: false });
+  const recs = data?.recommendations || [];
+  const page = filters.offset / filters.limit + 1;
+  const [filteredRows, setFilteredRows] = useState([]);
+  const [displayedRows, setDisplayedRows] = useState([]);
+
+  useEffect(() => {
+    setDisplayedRows(buildDisplayedRows(filteredRows));
+  }, [filteredRows, filters.limit, filters.offset]);
+
+  useEffect(() => {
+    const newFilteredRows = buildFilteredRows(recs, filters);
+    const newDisplayedRows = buildDisplayedRows(newFilteredRows);
+    setFilteredRows(newFilteredRows);
+    setDisplayedRows(newDisplayedRows);
+  }, [recs, filters]);
+
+  // constructs array of rows (from the initial data) checking currently applied filters
+  const buildFilteredRows = (allRows, filters) => {
+    const checkFilters = (rule) => {
+      return Object.entries(filters).every(([filterKey, filterValue]) => {
+        switch (filterKey) {
+          case 'text':
+            return rule.description
+              .toLowerCase()
+              .includes(filterValue.toLowerCase());
+          case FC.total_risk.urlParam:
+            return filterValue.includes(String(rule.total_risk));
+          case FC.category.urlParam:
+            return rule.tags.find((c) =>
+              filterValue.includes(String(RULE_CATEGORIES[c]))
+            );
+          case FC.impact.urlParam:
```

Comment:
I would rename FC to filterCategories, otherwise, anyone new looking at the code would have to hover over the code and check what is it exactly doing. Especially when we leaving RULE_CATEGORIES as it is

Reply:
✅